src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -9,11 +9,13 @@
 
 public class ValuesResolver {
 
-
     public Map<String, Object> getVariableValues(GraphQLSchema schema, List<VariableDefinition> variableDefinitions, Map<String, Object> inputs) {
         Map<String, Object> result = new LinkedHashMap<String, Object>();
         for (VariableDefinition variableDefinition : variableDefinitions) {
-            result.put(variableDefinition.getName(), getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName())));
+            Object variableValue = getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName()));
+            if (variableValue != null) {
+                result.put(variableDefinition.getName(), variableValue);
+            }
```

Comment:
I am wondering what the compatibility implications are here. I can't imagine any, but I can't imagine many things :).

Related to the above question: Why would you want this to be the way this works? I mean, this method returns the resulting `Map`, and the client then `get()`s from that `Map`. With our without the PR, they'd get `null` for missing keys. Right?